index.d.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Check if `input` is a ES2015 promise.
+ *
+ * @param input - Value to be checked.
+ *
+ * @example
+ *
+ * import isPromise from 'p-is-promise';
+ *
+ * isPromise(Promise.resolve('🦄'));
+ * //=> true
+ */
+export default function(input: unknown): input is Promise<unknown>;
```

Comment:
I think this should use generics:

```ts
export default function<T>(input: T): input is Promise<T>;
```

Reply:
Not really. The reason is that we don't know what type `input` would be. That's the reason why `isPromise()` exists. That's also why we decided not to do this in [is](https://github.com/sindresorhus/is/blob/master/source/index.ts#L152).

Currently when doing `isPromise<string>`, it enforces the `input` param to be of type `string`, which it clearly isn't because we don't know. So `input` should always be of type `unknown`.

What we first did in `is` is the following

```ts
export default function<T>(input: unknown): input is Promise<T>;
```

So you could do `if (isPromise<string>(foo)) {` and then TypeScript knew that the result of the promise was of type `string`. But that's incorrect, because `isPromise` does not detect the type of the result of the promise. So either the user should cast the result themselves, or use `typeof` to detect it. That's also the nice thing about using `unknown`, TypeScript enforces you to check it.

Reply:
Ah, makes sense.